### DISCUSSION TITLE
Default on startup to applying all updates before triggering subscriptions

### DISCUSSION THREAD

**Discussion by KyleAMathews** [electric-sql/electric#1032] [Bug reports]
Right now if you visit an app that's had a lot of updates, you see a long series of fast re-renders while changes stream in.

This is slow and undesirable. Instead there should just be a batch update once all updates have arrived and then trigger just one re-render.

**Comment by balegas** [comment]
This might be one of the oldest things in our todo list :D. 
- We need to have some additional sync state info to know that we're catching up with the server
- Batching updates would speedup the process a lot

**Reply by KyleAMathews** [reply]
Haha I'll bet!

**Comment by KyleAMathews** [comment]
👋 we've been working the last month on a rebuild of the Electric server over at a temporary repo https://github.com/electric-sql/electric-next/

You can read more about why we made the decision at https://next.electric-sql.com/about

We're really excited about all the new possibilities the new server brings and we hope you'll check it out soon and give us your feedback.

We're now moving the temporary repo back here. As part of that migration we're closing all the old issues, PRs, and discussions. We really appreciate you taking the time to start this discussion! If it's still relevant in the new system, please reopen.
